client/src/components/functionality-export-popup.vue

```diff
@@ -63,30 +88,65 @@ export default {
       }
       let idsToExport = []
       this.displayedFunctionalities.forEach(f => idsToExport.push(f.id))
+      let urlArgs = `exportType=${this.exporterId}&functionalities=${idsToExport.join(',')}`
+      if (this.fieldValue !== {}) {
+        let keys = Object.keys(this.fieldValue)
+        for (var idx in keys) {
+          let key = keys[idx]
+          let element = this.fieldValue[key]
+          urlArgs = `${urlArgs}&${key}=${element}`
+        }
+      }
       Vue.http
-        .get(api.paths.functionalities(this.projectCode) + `/export?exportType=${this.exporterId}&functionalities=${idsToExport.join(',')}`, api.REQUEST_OPTIONS)
+        .get(api.paths.functionalities(this.projectCode) + `/export?${urlArgs}`, api.REQUEST_OPTIONS)
         .then((response) => {
-          this.triggerDownload(response.bodyText)
+          console.log({ r: response })
+          this.triggerDownload(response)
           this.exporterId = ''
         }, (error) => {
           api.handleError(error)
         })
     },
 
-    triggerDownload (content) {
-      const blob = new Blob([content], { type: 'text/plain' })
+    triggerDownload (response) {
+      let content = this.decodeB64Stream(response.bodyText)
+      const blob = new Blob([content], { type: 'application/octet-stream' })
       const event = document.createEvent('MouseEvents')
       let anchor = document.createElement('a')
-      anchor.download = `exportAraCartography-${this.$route.params.projectCode}.json`
+      let exporterFormat = ''
+      for (let idx in this.availableExporters) {
+        if (this.availableExporters[idx].id === this.exporterId) {
+          exporterFormat = this.availableExporters[idx].format
+        }
+      }
+      anchor.download = `exportAraCartography-${this.$route.params.projectCode}.${exporterFormat}`
       anchor.href = window.URL.createObjectURL(blob)
-      anchor.dataset.downloadurl = ['text/json', anchor.download, anchor.href].join(':')
       event.initEvent('click', true, false, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null)
       anchor.dispatchEvent(event)
     },
 
     cancelExport () {
       this.exporterId = ''
+      this.fieldValue = {}
+    },
+
+    changeCurrentExporter (id) {
+      let currentExporter = this.availableExporters.filter((ex) => ex.id === id)[0]
+      this.requiredFields = (currentExporter) ? currentExporter.requiredFields : undefined
+      this.selectedExporterId = id
+      this.fieldValue = {}
+      return id
+    },
+
+    decodeB64Stream (contentBase64) {
+      let binaryStr = window.atob(contentBase64)
+      let bStrLen = binaryStr.length
+      let contentArray = new Uint8Array(bStrLen)
```

Comment:
In my opinion, the suffix `array` is not required: the variable name shouldn't care about what type (e.g. array, list, etc.) you use to store your data, rather `contents` (plural) could suffice.

Reply:
I don't agree with you on this point. Here we manipulate a string by using directly the bytes in it.

When I use `contentArray` it's not for "an array of contents" but the "content in bytes array format", if you want I can change it to `contentByteArray`